apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx

```diff
@@ -0,0 +1,142 @@
+  const WidthComponent = ({ startColor, middleColor, finalColor }: TestCase) => {
+    const colorActiveSV = useSharedValue(startColor);
+    const colorPassiveSV = useSharedValue(startColor);
+
+    const refOne = useTestRef(Component.ACTIVE);
+    const refTwo = useTestRef(Component.PASSIVE);
+
+    const styleActive = useAnimatedStyle(() => {
+      return {
+        backgroundColor: withSequence(
+          withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
+          withDelay(DELAY, withTiming(middleColor, { duration: 300 })),
+          withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
+        ),
+      };
+    });
+    const stylePassive = useAnimatedStyle(() => {
+      return {
+        backgroundColor: colorPassiveSV.value,
+      };
+    });
+
+    useEffect(() => {
+      colorActiveSV.value = finalColor;
+    }, [colorActiveSV, finalColor]);
+
+    useEffect(() => {
+      colorPassiveSV.value = withSequence(
+        withDelay(DELAY, withTiming(finalColor, { duration: 200 })),
+        withDelay(DELAY, withTiming(middleColor, { duration: 300 })),
+        withDelay(DELAY, withTiming(finalColor, { duration: 200 })),
+      );
+    }, [colorPassiveSV, finalColor, middleColor]);
+
+    return (
+      <View style={styles.container}>
+        <Animated.View ref={refOne} style={[styles.animatedBox, styleActive]} />
+        <Animated.View ref={refTwo} style={[styles.animatedBox, stylePassive]} />
+      </View>
+    );
+  };
+
+  test.only.each([
+    {
+      startColor: 'gold',
+      startColorHex: '#ffd700ff',
+      middleColor: 'forestgreen',
+      middleColorHex: '#228b22ff',
+      finalColor: 'darkblue',
+      finalColorHex: '#00008bff',
+    },
+    {
+      startColor: 'gold',
+      startColorHex: '#ffd700ff',
+      middleColor: 'hsl(180, 50%, 50%)',
+      middleColorHex: '#40bfbfff',
+      finalColor: 'hsl(120,100%,50%)',
+      finalColorHex: '#00ff00ff',
+    },
+    {
+      startColor: 'gold',
+      startColorHex: '#ffd700ff',
+      middleColor: 'hsl(70, 100%, 75%)',
+      middleColorHex: '#eaff80ff',
+      finalColor: 'hsl(120,100%,50%)',
+      finalColorHex: '#00ff00ff',
+    },
+    {
+      startColor: 'hwb(70, 50%, 0%)',
+      startColorHex: '#eaff80ff',
+      middleColor: 'hsl(180, 50%, 50%)',
+      middleColorHex: '#40bfbfff',
+      finalColor: 'hsl(120,100%,50%)',
+      finalColorHex: '#00ff00ff',
+    },
+  ])(
+    'Animate ${startColor} → ${finalColor} → ${middleColor} → ${finalColor}',
+    async ({ startColor, startColorHex, middleColor, middleColorHex, finalColor, finalColorHex }) => {
+      await render(<WidthComponent startColor={startColor} middleColor={middleColor} finalColor={finalColor} />);
+      const activeComponent = getTestComponent(Component.ACTIVE);
+      const passiveComponent = getTestComponent(Component.PASSIVE);
+
+      await wait(DELAY / 2);
+      // TODO Decide what should be the starting value of activeComponent
```

Comment:
Do you have some a list of issues that you found?

Reply:
As you can see there are two components that should behave the same way, but they don't:
```tsx
expect(await activeComponent.getAnimatedStyle('backgroundColor'))
    .not.toBe(startColorHex, ComparisonMode.COLOR);
expect(await passiveComponent.getAnimatedStyle('backgroundColor'))
    .toBe(startColorHex, ComparisonMode.COLOR);
```

The problem with the activeComponent that the following code:
```diff
!   const colorActiveSV = useSharedValue(startColor);

    const styleActive = useAnimatedStyle(() => {
      return {
        backgroundColor: withSequence(
!         withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
          withDelay(DELAY, withTiming(middleColor, { duration: 300 })),
          withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
        ),
      };
    });

    useEffect(() => {
!     colorActiveSV.value = finalColor;
    }, [colorActiveSV, finalColor]);
 ```
 Should produce the animation: startColor → finalColor → middleColor → finalColor,
 but we get finalColor → middleColor → finalColor

Reply:
Thats the way it looks:

https://github.com/software-mansion/react-native-reanimated/assets/56199675/fe692b5e-d9f9-4c24-8639-580d7c0f541d

Reply:
I was checking several times if this should be considered a bug and I think it should - when we have similar ways to animate with other animations, we get the bottom result,